packages/@aws-cdk/aws-iam/lib/role.ts

```diff
@@ -123,50 +123,64 @@ export interface RoleProps {
   readonly maxSessionDuration?: Duration;
 }
 
+/**
+ * Options allowing customizing the behavior of {@link Role.fromRoleArn}.
+ */
+export interface FromRoleArnOptions {
+  /**
+   * Whether the imported role allows "mutating" operations.
+   * Of course, we cannot change a resource existing outside the CDK,
```

Comment:
This line is already very telling that the name is actually not great.

How about `canModifyPolicy` or `modifyPolicy` or `addPolicies`?

Reply:
I actually think the name of the property is fine, but my comment is the bad thing here.

I was too influenced by the way CloudFormation models these things - by having a `Policy` resource that has a list of `Role`s it is attached to. But I believe that is the wrong way to think about it. Policy resources by themselves are useless. Only through attaching them to roles do they do anything. So in my opinion, we are absolutely mutating something that exists outside the CDK.

I've updated the wording, let me know if you agree with it now.

Reply:
Can you at least mark the property `@experimental` then, so we can at least come back to the name at some point in the future?

Reply:
I don't really like that - I'd rather deprecate this property and add a properly-named one in the future. But sure, added.